enigma-swing/src/test/java/cuchaz/enigma/PackageRenameTest.java

```diff
@@ -0,0 +1,144 @@
+package cuchaz.enigma;
+
+import cuchaz.enigma.gui.EditableType;
+import cuchaz.enigma.gui.Gui;
+import cuchaz.enigma.gui.docker.AllClassesDocker;
+import cuchaz.enigma.gui.docker.Docker;
+import cuchaz.enigma.gui.elements.ClassSelectorPopupMenu;
+import cuchaz.enigma.translation.TranslateResult;
+import cuchaz.enigma.translation.Translator;
+import cuchaz.enigma.translation.mapping.serde.MappingFormat;
+import cuchaz.enigma.translation.representation.entry.Entry;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.condition.DisabledIf;
+
+import java.nio.file.Path;
+import java.util.EnumSet;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.CountDownLatch;
+
+import static cuchaz.enigma.TestEntryFactory.*;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.notNullValue;
+
+public class PackageRenameTest {
+	public static final Path JAR = TestUtil.obfJar("complete");
+	public static final Path MAPPINGS = Path.of("src/test/resources/test_mappings");
+	private static Translator deobfuscator;
+
+	@Test
+	@DisabledIf(value = "java.awt.GraphicsEnvironment#isHeadless", disabledReason = "headless environment")
```

Comment:
The exception may be applied to the whole class instead of individual tests, so we don't have to copy the exact same values every time we want a new test